File: 7_reverse_integer/reverse_integer.c
int reverse(int x) {
    int result = 0;
    char flag = x > 0 ? 1 : -1;
    if (x <= -pow(2, 31)) {
        return 0;
    }
    x = x > 0 ? x : -x;
    while (x > 0 || x % 10 > 0) {
        if ((flag == 1 && result > (pow(2, 31) - 1 - x % 10) / 10 )  ||
            (flag == -1 && result > (pow(2, 31)- x % 10) / 10)) {
            return 0;
        }
        result = result * 10 + x % 10;
        x = x / 10;
    }
    return result * flag;
}

File: 7_reverse_integer/reverse_integer.py
class Solution:
    def reverse(self, x: int) -> int:
        result = 0
        flag = 1 if x >= 0 else - 1
        x = abs(x)
        while x > 0 or x % 10:
            result = result * 10 + x % 10
            x //= 10
        if result < -2 ** 31 or result > 2 ** 31 - 1:
            return 0
        return flag * result


